### ISSUE TITLE
Cannot build tModLoader mod from VS on ARM64 Windows system

### ISSUE THREAD

**Issue by WoodStuff** [bluemagic123/tModLoader#4335]
### Version

Latest 1.4.4-stable

### OS

Windows

### Platform

Steam

### This bug affects...

Mod capability as a Modder

### Description

On a Windows 11 ARM64 system, when I click the run button in Visual Studio 2022, it errors saying that it found a 32-bit SDK, but I have a 64-bit SDK.

### Steps to reproduce

1. Have an ARM64 Windows system (not sure if it has to be Windows 11) with Visual Studio 2022 and a tModLoader mod in development
2. Open the mod in VS2022 and click the Terraria button with a green arrow
3. Terraria refuses to open because of the error

### Expected Behavior

tModLoader opens with the mod on

### Actual Behavior

tModLoader doesn't open and it errors

### Reproduction frequency

Everytime

### Additional Information

I've asked for help with this issue in the discord server and it was confirmed this is a bug, it might be related to this piece of code:
![It might be related to this piece of code](https://github.com/user-attachments/assets/9b7d217f-8880-43fb-9e5c-a38575398e9c)

**Comment by JavidPack** [comment]
Are you saying that tModLoader works aside from when doing it though VS on your ARM system? tModLoader itself shouldn't work at all on ARM computers, but I haven't had any experience with the new windows ARM computers, I think they have an emulation process that makes it work, but I don't know exactly how that changes how the game is informed about the processor architecture.

If it's just VS that is erroring, my guess is it's running it as a native arm application, I don't know how to avoid that right now. 

I've attached a custom build that changes the check to check specifically for x86 instead of not x64, replace the tModLoader.dll in your install directory with it and try rebuilding an debugging again. Let me know if it works. If it doesn't, please post the logs next time.

[tModLoader.zip](https://github.com/user-attachments/files/16514481/tModLoader.zip)

**Comment by WoodStuff** [comment]
yeah its just visual studio that doesnt want to start tmodloader i can start tml normally from steam, as for the dll replacing it makes tml unable to start even by steam: 
![image](https://github.com/user-attachments/assets/b5589fc2-2bcb-48cd-962f-347d9802a95f)
these are the logs with the replaced dll:
client.log: https://pastebin.com/JSZJMxJ6
environment-client.log: https://pastebin.com/1TLiBNEj
Launch.log: https://pastebin.com/nvqJt6jj
Natives.log is empty
as for trying to start the game through visual studio heres the output https://pastebin.com/BMyqbzHr

**Comment by JavidPack** [comment]
Ah, sorry, I made a mistake with the build I sent you. Try this new build. Also, post the new logs as the actual files (just drag and drop them into the text box), not pastebin links. 

[tModLoader-platformcheck_B.zip](https://github.com/user-attachments/files/16552548/tModLoader-platformcheck_B.zip)

**Comment by WoodStuff** [comment]
actually even building from vs doesnt work, so its probably not playing tml from vs thats causing the issue then
in any case here are the logs
[output.txt](https://github.com/user-attachments/files/16560189/output.txt) (visual studio output)
these havent updated when i tried to run tml from visual studio but they exist i guess
[client.log](https://github.com/user-attachments/files/16560204/client.log)
[environment-client.log](https://github.com/user-attachments/files/16560205/environment-client.log)
[Launch.log](https://github.com/user-attachments/files/16560206/Launch.log)
[Natives.log](https://github.com/user-attachments/files/16560207/Natives.log)
[terrariasteamclient.log](https://github.com/user-attachments/files/16560208/terrariasteamclient.log)

**Comment by WoodStuff** [comment]
also this error appears in the error list after i build, idk if its relevant but im posting it here if it is
![image](https://github.com/user-attachments/assets/6ff257da-0773-4e33-97a8-fc19da50dff2)

**Comment by JavidPack** [comment]
I can't open that output.txt, something weird with github I guess. Can you send that again just as text.

**Comment by WoodStuff** [comment]
sorry i forgot to save the file so it was blank im dumb lol, this should work now
[output.txt](https://github.com/user-attachments/files/16569125/output.txt)

**Comment by JavidPack** [comment]
Sorry, I didn't see this comment earlier.

I have something for you to try that I think might work:

1. Open up `C:\Program Files (x86)\Steam\steamapps\common\tModLoader\tMLMod.targets` in a text editor.
2. Scroll down to the bottom and find the `<Exec Command="dotnet $(tMLServerPath) -build $(ProjectDir) -eac $(TargetPath) -define &quot;$(DefineConstants)&quot; -unsafe $(AllowUnsafeBlocks) $(ExtraBuildModFlags)" WorkingDirectory="$(tMLSteamPath)"/>` line.
3. Replace `dotnet` with `&quot;C:\Program Files (x86)\Steam\steamapps\common\tModLoader\dotnet\dotnet.exe&quot;`. The final line should read `<Exec Command="&quot;C:\Program Files (x86)\Steam\steamapps\common\tModLoader\dotnet\dotnet.exe&quot; $(tMLServerPath) -build $(ProjectDir) -eac $(TargetPath) -define &quot;$(DefineConstants)&quot; -unsafe $(AllowUnsafeBlocks) $(ExtraBuildModFlags)" WorkingDirectory="$(tMLSteamPath)"/>`
4. Save the file
5. Open up `C:\Users\woods\Documents\My Games\Terraria\tModLoader\ModSources\Eggpack\Properties\launchSettings.json` in a text etitor.
6. Find the `"executablePath": "$(DotNetName)",` line in the `Terraria` section. Should be line 5.
7. Change `"executablePath": "$(DotNetName)",` to `"executablePath": "C:\\Program Files (x86)\\Steam\\steamapps\\common\\tModLoader\\dotnet\\dotnet.exe",`
8. Save that file
9. Attempt to Build the mod in VS, it should hopefully work.
10. Attempt to debug the mod in VS, it should also hopefully work.

If this works, we can look into fixing this more generally, I just wanted to verify if this would fix the issue or not. Again, sorry about the delay.

**Comment by WoodStuff** [comment]
it works! i can both build and debug it
